# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1424151745 18000
#      Tue Feb 17 00:42:25 2015 -0500
# Node ID 3c3b9fce7d2b28740671a4e7b9a270d97cba73c3
# Parent  94cc1bdaf8841ff50ccdb9a89c6328ddc5d8991f
improve compatibility of sprintf function.

* file-io.cc (Fsprintf): Return 1x0 empty string if no conversion
occurs.

diff --git a/libinterp/corefcn/file-io.cc b/libinterp/corefcn/file-io.cc
--- a/libinterp/corefcn/file-io.cc
+++ b/libinterp/corefcn/file-io.cc
@@ -1091,18 +1091,23 @@ converted.\n\
                   tmp_args.resize (nargin-1, octave_value ());
 
                   for (int i = 1; i < nargin; i++)
                     tmp_args(i-1) = args(i);
                 }
 
               retval(2) = os.printf (fmt_arg, tmp_args, who);
               retval(1) = os.error ();
-              retval(0) = octave_value (ostr->str (),
-                                        fmt_arg.is_sq_string () ? '\'' : '"');
+
+              std::string result = ostr->str ();
+              char type = fmt_arg.is_sq_string () ? '\'' : '"';
+
+              retval(0) = (result.empty ()
+                           ? octave_value (charMatrix (1, 0), type)
+                           : octave_value (result, type));
             }
           else
             ::error ("%s: format TEMPLATE must be a string", who.c_str ());
         }
       else
         ::error ("%s: unable to create output buffer", who.c_str ());
     }
   else
